### PR TITLE
Support filtering and labeling by relationship type for item-piechart

### LINKED ISSUE #270: item-piechart: filter by relationship between source and target
Add the :type: attribute in the same way as for item-matrix

### PR DESCRIPTION
**Pull request by JasperCraeghs**
- Added `:sourcetype:` option to the `item-piechart` directive
- Added `:targettype:` option to the `item-piechart` directive
- Fixed a bug that resulted in using the `<<attribute>>` value of the nested target item that was parsed last as label instead of respecting the order of the arguments of the `:<<attribute>>:` option, or, in case of multiple target items, discarding the effect of those defined first.

Resolves #270

### CODE REVIEW

**Comment by JasperCraeghs** [comment]
Which label shall be used  for an RQT-item that is linked to two TEST-items, of which only one is linked to a TEST_REP-item? `covered` or `executed`? 

```
.. item-piechart::
    :id_set: RQT TEST TEST_REP
    :label_set: uncovered, covered, executed
```

**Comment by bavovanachte** [comment]
I would say "covered" specifically in the case of test coverage. I haven't used the item-piechart directive yet, but from the documentation, I understand that "covered" in this case would mean that the REQ has a TEST linked, but not a test report. 

I would think it's deceiving to mark a requirement as "executed" if only one of its testcases has been run. I'm guessing requirements with multiple testcases is not uncommon, and my opinion would be that a requirement is only covered if all of its planned testcases have been executed

**Comment by Juegadamas** [comment]
I'm with @bavovanachte. I'd say covered for the same reasons.

File: doc/usage.rst [hunk 1]
```diff
@@ -604,12 +606,29 @@ are comma-separated lists.
...
+
+:targettype: *optional*, *multiple arguments (space-separated)*
+
+    The list of relationships that should be used to filter the nested target. The relationships considered for
+    filtering are from the "Target" items to the "Target-of-target" items. These relationhips will also be used to label
+    additional slices if the *<<attribute>>* option that accepts multiple arguments is unused.
+    In this example, if a TEST-item is not linked to a TEST_REP-item with one or more of
+    *passed_by/skipped_by/failed_by*, the source item will be labeled as *covered* instead of
+    *passes*, *skipped* or *fails*, which are the human readable and reversed
+    forms of the arguments for this option. If an RQT-item with multiple TEST-items, one linked to
```
Comment (Letme) [inline doc/usage.rst:624]: This sentence is incomplete

Comment (JasperCraeghs) [reply doc/usage.rst]: b58d73fcd56277b33de064033fef52ed4c799edd

**Pull request merged**